devices/ICub/src/ICub.cpp

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright (C) 2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include "ICub.h"
+
+#include <yarp/os/BufferedPort.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/os/Network.h>
+
+#include <assert.h>
+#include <map>
+#include <mutex>
+#include <string>
+#include <vector>
+
+using namespace wearable::devices;
+
+const std::string LogPrefix = "ICub :";
+
+class WrenchPort : public yarp::os::BufferedPort<yarp::os::Bottle>
+{
+public:
+    std::vector<double> wrenchValues;
+    bool firstRun = true;
+    mutable std::mutex mtx;
+
```

Comment:
Using the Network member was throwing compile-time error on Mac OS. So, it was removed in https://github.com/robotology-playground/wearables/commit/c0669c49ab749f81f81921fcf49ec322675dfd49. @lrapetti please add the details if you have any error messages.

Reply:
I removed that part in c0669c4 because otherwise in `MacOS` we were getting the following error:
```
[ 86%] Linking CXX shared module ../../lib/yarp/ICub.so
Undefined symbols for architecture x86_64:
  "yarp::os::Network::init()", referenced from:
      wearable::devices::ICub::open(yarp::os::Searchable&) in ICub.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/yarp/ICub.so] Error 1
make[1]: *** [devices/ICub/CMakeFiles/ICub.dir/all] Error 2
make: *** [all] Error 2
```

If I try to add a `yarp::os::Network` member, again I fall in similar error:
```
[ 86%] Linking CXX shared module ../../lib/yarp/ICub.so
Undefined symbols for architecture x86_64:
  "yarp::os::Network::init()", referenced from:
      wearable::devices::ICub::open(yarp::os::Searchable&) in ICub.cpp.o
  "yarp::os::Network::Network()", referenced from:
      WrenchPort::WrenchPort() in ICub.cpp.o
  "yarp::os::Network::~Network()", referenced from:
      WrenchPort::~WrenchPort() in ICub.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/yarp/ICub.so] Error 1
make[1]: *** [devices/ICub/CMakeFiles/ICub.dir/all] Error 2
make: *** [all] Error 2
```

Reply:
I suspect that something was not linked correctly, can you try to link  `YARP::YARP_init` target?

Reply:
> I suspect that something was not linked correctly, can you try to link `YARP::YARP_init` target?

linking `YARP_YARP_init` it compiles correctly when adding `yarp::os::Network yarpNetwork;`

Reply:
Great!

Reply:
@Yeshasvitvs if you want I can push this change

Reply:
Done with https://github.com/robotology-playground/wearables/pull/25/commits/ee3990a7328856798b72dbcc790f4eee7e4255b0.